test/execution/atomics_test.cpp

```diff
@@ -0,0 +1,673 @@
+#include <random>
+#include <string>
+
+#include "common/macros.h"
+#include "common/worker_pool.h"
+#include "execution/ast/ast_node_factory.h"
+#include "execution/ast/context.h"
+#include "execution/compiler/codegen.h"
+#include "execution/compiler/compiler.h"
+#include "execution/compiler/function_builder.h"
+#include "execution/sema/error_reporter.h"
+#include "execution/tpl_test.h"
+#include "execution/util/region.h"
+#include "execution/util/region_containers.h"
+#include "execution/vm/module.h"
+#include "execution/vm/vm_defs.h"
+#include "test_util/multithread_test_util.h"
+
+namespace noisepage::execution::test {
+
+class AtomicsTest : public TplTest {
+ public:
+  AtomicsTest() : region_("atomics_test"), pos_() {}
+
+  util::Region *Region() { return &region_; }
+
+  const SourcePosition &EmptyPos() const { return pos_; }
+
+  util::Region region_;
+  SourcePosition pos_;
+};
+
+// NOLINTNEXTLINE
+TEST_F(AtomicsTest, AtomicAndOr1) {
+  // Setup the compilation environment
+  sema::ErrorReporter error_reporter(&region_);
```

Comment:
Is there a reason you're handcrafting the AST rather than using raw TPL source and running it through the compilation pipeline?

There's a bit of code duplication that might benefit from a refactor. Also, add some comments between the test that, at least, show the text you intend to generate.

Reply:
Does the main binary use a different parser than the standalone `tpl` utility?  The standalone utility's parser was crashing and burning (re: previous Slack conversation) so I assumed that the same TPL would not work here.  I would absolutely prefer for it to be plain TPL as that is more sustainable.

Reply:
Yes, it is the same parser. IIRC, it was crashing when you try to dereference. But, since you're now using a builtin function, it should work, right?